src/Core/Form/IdentifiableObject/CommandBuilder/Accessor/CommandAccessor.php

```diff
@@ -0,0 +1,165 @@
+ */
+class CommandAccessor
+{
+    /**
+     * @var CommandAccessorConfig
+     */
+    private $config;
+
+    /**
+     * @var PropertyAccessorInterface
+     */
+    private $propertyAccessor;
+
+    public function __construct(CommandAccessorConfig $config)
+    {
+        $this->config = $config;
+        $this->propertyAccessor = PropertyAccess::createPropertyAccessorBuilder()
+            ->enableExceptionOnInvalidIndex()
+            ->enableExceptionOnInvalidPropertyPath()
+            ->disableMagicCall()
+            ->getPropertyAccessor()
+        ;
+    }
+
+    /**
+     * If only single store command is provided then only this one will be prefilled, however if a command for all
+     * stores is also provided then this component will check in the data (for each field) if the modification targets
+     * all the stores, or not, and prefill the appropriate command accordingly.
+     *
+     * @param array $data
+     * @param mixed $singleStoreCommand
+     * @param mixed|null $allStoresCommand
+     *
+     * @return array Returns modified commands (if no field was detected an empty array is returned)
+     */
+    public function buildCommands(
+        array $data,
+        $singleStoreCommand,
+        $allStoresCommand = null
+    ): array {
+        $modifiedCommands = [];
+        foreach ($this->config->getFields() as $commandField) {
+            try {
+                $value = $this->propertyAccessor->getValue($data, $commandField->getDataPath());
+                $castedValue = $this->castValue($value, $commandField->getType());
+
+                $command = $this->getAppropriateCommand($data, $commandField, $singleStoreCommand, $allStoresCommand);
+                $this->propertyAccessor->setValue($command, $commandField->getCommandSetter(), $castedValue);
+                if (!in_array($command, $modifiedCommands)) {
+                    $modifiedCommands[] = $command;
+                }
+            } catch (NoSuchIndexException $e) {
+                // Data has no value for this field
+            }
+        }
+
+        // Make sure the order of returned commands is always consistent (single store comes first)
+        $commands = [];
+        if (in_array($singleStoreCommand, $modifiedCommands)) {
+            $commands[] = $singleStoreCommand;
+        }
+        if (in_array($allStoresCommand, $modifiedCommands)) {
+            $commands[] = $allStoresCommand;
+        }
+
+        return $commands;
+    }
+
+    /**
+     * Check if the data has a mapping checkbox to modify all stores for the tested field, if so use the allStoresCommand
+     *
+     * @param array $data
+     * @param CommandField $commandField
+     * @param mixed $singleStoreCommand
+     * @param mixed|null $allStoresCommand
+     *
+     * @return mixed
+     */
+    private function getAppropriateCommand(
+        array $data,
+        CommandField $commandField,
+        $singleStoreCommand,
```

Comment:
same as previous, can be typed I think

Reply:
I can't type this because it needs to work with any CQRS command, `UpdateProductPricesCommand`, `UpdateProductBasicInformationCommand`, ...

They share no common class nor interface so I can't type it, `mixed` can't be used as a strong type either and I don't belive there is a generic `Object` in PHP